src/basic_recipes/datashader.jl

```diff
@@ -0,0 +1,388 @@
+    @assert eltype(aggbuffer) === typeof(null(op))
+
+    # using ReshapedArray directly like this is not advised, but as it lives only briefly it should be ok
+    out = Base.ReshapedArray(aggbuffer, (xsize, ysize, Threads.nthreads()), ())
+    out2 = Base.ReshapedArray(pixelbuffer, (xsize, ysize), ())
+    n = length(points)
+    chunks = round.(Int, range(1, n, length = Threads.nthreads()+1))
+
+    @threads for t in 1:Threads.nthreads()
+        from = chunks[t]
+        to = chunks[t+1]
+        for idx in from:to
+            p = @inbounds point_func(points[idx])
+            x = p[1]
+            y = p[2]
+            if length(p) > 2 # should compile away
+                z = p[3]
+            end
+            xmin ≤ x ≤ xmax || continue
+            ymin ≤ y ≤ ymax || continue
+            i = 1 + floor(Int, xscale*(x-xmin))
+            j = 1 + floor(Int, yscale*(y-ymin))
+            if length(p) == 2 # should compile away
+                @inbounds out[i,j,t] = update(op, out[i,j,t])
+            elseif length(p) == 3
+                @inbounds out[i,j,t] = update(op, out[i,j,t], z)
+            end
+        end
+    end
+    # reduce along the thread dimension
+    mini, maxi = Inf, -Inf
+    for j in 1:ysize
+        for i in 1:xsize
+            @inbounds val = out[i,j,1]
+            for t in 2:Threads.nthreads()
+                @inbounds val = merge(op, val, out[i,j,t])
+            end
+            # update the value in out2 directly in this loop
+            final_value = local_op(value(op, val))
+            if isfinite(final_value)
+                mini = min(final_value, mini)
+                maxi = max(final_value, maxi)
+            end
+            @inbounds out2[i, j] = final_value
+        end
+    end
+    return (mini, maxi)
+end
+
+
+export AggAny, AggCount, AggMean, AggSum, AggSerial, AggThreads
+
+end
+
+using ..PixelAggregation
+
+function equalize_histogram(matrix; nbins=256 * 256)
+    h_eq = StatsBase.fit(StatsBase.Histogram, vec(matrix); nbins=nbins)
+    h_eq = normalize(h_eq; mode=:density)
+    cdf = cumsum(h_eq.weights)
+    cdf = cdf / cdf[end]
+    edg = h_eq.edges[1]
+    # TODO is this the correct linear interpolation?
+    return Makie.interpolated_getindex.((cdf,), matrix, (Vec2f(first(edg), last(edg)),))
+end
+
+"""
+    datashader(points::AbstractVector{<: Point})
+
+Points can be any array type supporting iteration & getindex, including memory mapped arrays.
+If you have x + y coordinates seperated and want to avoid conversion + copy, consider using:
+```Julia
+using Makie.StructArrays
+points = StructArray{Point2f}((x, y))
+datashader(points)
+```
+Do pay attention though, that if x and y don't have a fast iteration/getindex implemented, this might be slower then just copying it into a new array.
+
+For best performance, use `method=Makie.AggThreads()` and make sure to start julia with `julia -t8` or have the environment variable `JULIA_NUM_THREADS` set to the number of cores you have.
```

Comment:
`-t8` seems like a recipe for problems as general advice :) Doesn't Julia have auto-threads now?